### PR TITLE
388 key details on hover

### PR DESCRIPTION
**Pull request by AliciadW**
screenshot 2019-02-26 at 10 59 50

Details of work is shown in popover on hover, user can click through to linked works from popover

### CODE REVIEW

**Comment by longhotsummer** [comment]
I think this is a great proof of concept. To finalise:

1. @goose-life please can you add a work-based url such as `/works/.../popup` which renders the chunk of HTML to show in the popover. Don't use `{% extends ... %}` so that it doesn't inherit all the placeholder html.

2. @AliciadW can we set it up so that the content is loaded dynamically from the above endpoint on mouseover, rather than injected on the server? See eg my comment in https://github.com/laws-africa/indigo/issues/388 -- we're going to want this to popup on a number of links on the page and don't want to have to duplicate it.

3. @AliciadW the JS code can live in https://github.com/laws-africa/indigo/blob/master/indigo_app/static/javascript/indigo/utils.js

**Comment by longhotsummer** [comment]
This is a bit outdated but is very similar to what we need: https://stackoverflow.com/questions/8130069/load-a-bootstrap-popover-content-with-ajax-is-this-possible

File: indigo_app/static/javascript/indigo/utils.js [hunk 1]
```diff
@@ -15,3 +15,21 @@ $(function() {
 
   $('body').on('click', 'a[data-confirm], button[data-confirm]', handleConfirm);
 });
+
+// Show popover when hovering on selected links
+$('body').on('mouseenter', 'a[data-popup-url]', function() {
+  var _this = this;
+  $.get(this.href + 'popup').then(function(html) {
```
Comment (longhotsummer) [inline indigo_app/static/javascript/indigo/utils.js]: This URL should be the contents of the `data-popup-url` attribute.

File: indigo_app/static/javascript/indigo/utils.js [hunk 2]
```diff
@@ -15,3 +15,21 @@ $(function() {
...
+    $(_this).popover({content: html, html: true}).popover('show');
+  });
+  $('.popover').mouseleave(function () {
+    $(_this).popover('hide');
+  });
+}).on('mouseleave','a[href^="/works/"]:not(.btn)', function(){
```
Comment (longhotsummer) [inline indigo_app/static/javascript/indigo/utils.js]: The scoping on this event should be `a[data-popup-url]` I think?

File: indigo_app/static/javascript/indigo/utils.js [hunk 3]
```diff
@@ -15,3 +15,21 @@ $(function() {
 
   $('body').on('click', 'a[data-confirm], button[data-confirm]', handleConfirm);
 });
+
+// Show popover when hovering on selected links
+$('body').on('mouseenter', 'a[data-popup-url]', function() {
```
Comment (longhotsummer) [inline indigo_app/static/javascript/indigo/utils.js]: Will this allow us to click on links in the popup?

File: indigo_app/static/javascript/indigo/utils.js (hunk 2 above)
Comment (AliciadW) [reply indigo_app/static/javascript/indigo/utils.js]: Yes, it should. I forgot to update this.

File: indigo_app/static/javascript/indigo/utils.js (hunk 3 above)
Comment (AliciadW) [reply indigo_app/static/javascript/indigo/utils.js]: Yes, you can still click on a link in the popup

File: indigo_app/static/javascript/indigo/utils.js (hunk 1 above)
Comment (AliciadW) [reply indigo_app/static/javascript/indigo/utils.js]: Updated

File: indigo_app/static/javascript/indigo/utils.js (hunk 3 above)
Comment (longhotsummer) [reply indigo_app/static/javascript/indigo/utils.js]: Great.

File: indigo_app/templates/indigo_api/_work_related_table.html [hunk 4]
```diff
@@ -14,7 +14,7 @@ <h5 class="card-header">{{ title }}</h5>
     {% for item in items %}
       <tr>
         <td><strong>{{ item.rel|capfirst }}</strong></td>
-        <td><a href="{% url 'work' item.work.frbr_uri %}">{{ item.work.title }}</a></td>
+        <td id="testing"><a href="{% url 'work' frbr_uri=item.work.frbr_uri %}" data-popup-url="{% url 'work_popup' frbr_uri=item.work.frbr_uri %}">{{ item.work.title }}</a></td>
```
Comment (longhotsummer) [inline indigo_app/templates/indigo_api/_work_related_table.html]: Can drop the id now :)

**Comment by longhotsummer** [comment]
How about some box-shadow on the popover element: `box-shadow: 4px 4px 8px 0px rgba(0, 0, 0, 0.25);`

File: indigo_app/templates/indigo_api/work_overview.html [hunk 5]
```diff
@@ -45,7 +45,7 @@ <h4 class="text-muted">{{ work.frbr_uri }}</h4>
               </div>
               {% if work.publication_document %}
               <div>
-                <i class="fas fa-file text-muted"></i> <a href="{% url 'work_publication_document' frbr_uri=work.frbr_uri filename=work.publication_document.filename %}" target="_blank" rel="noopener">{{ work.publication_document.filename }}</a> <span class="text-muted">({{ work.publication_document.size|filesizeformat }})</span>
+                <i class="fas fa-file text-muted"></i> <a href="{% url 'work_publication_document' frbr_uri=work.frbr_uri filename=work.publication_document.filename %}" target="_blank" rel="noopener" data-popup-url="{% url 'work_popup' frbr_uri=work.frbr_uri %}">{{ work.publication_document.filename }}</a> <span class="text-muted">({{ work.publication_document.size|filesizeformat }})</span>
```
Comment (goose-life) [inline indigo_app/templates/indigo_api/work_overview.html]:
```suggestion
                <i class="fas fa-file text-muted"></i> <a href="{% url 'work_publication_document' frbr_uri=work.frbr_uri filename=work.publication_document.filename %}" target="_blank" rel="noopener">{{ work.publication_document.filename }}</a> <span class="text-muted">({{ work.publication_document.size|filesizeformat }})</span>
```

**Pull request merged**